widgetssdk/src/main/java/com/glia/widgets/engagement/domain/OnIncomingEngagementRequestTimeoutUseCase.kt

```diff
@@ -0,0 +1,18 @@
+package com.glia.widgets.engagement.domain
+
+import com.glia.androidsdk.EngagementRequest.Outcome
+import com.glia.widgets.engagement.EngagementRepository
+import io.reactivex.rxjava3.core.Flowable
+
+internal interface OnIncomingEngagementRequestTimeoutUseCase {
+    operator fun invoke(): Flowable<Unit>
+}
+
+internal class OnIncomingEngagementRequestTimeoutUseCaseImpl(
+    private val engagementRepository: EngagementRepository
+) : OnIncomingEngagementRequestTimeoutUseCase {
+
+    override fun invoke(): Flowable<Unit> = engagementRepository.engagementOutcome
+        .filter { it == Outcome.TIMEOUT }
+        .map { }
```

Comment:
What is the benefit of converting this to `Unit` and then ignoring?
I think since we are ignoring the value, we can avoid this step.

Reply:
No, I mean we ignore the result when subscribing to this `Flowable`, so no need to map it to `Unit`